node/pkg/accountant/submit_obs.go

```diff
@@ -205,69 +304,73 @@ func SubmitObservationToContract(
 	txResp, err := wormchainConn.SignAndBroadcastTx(ctx, &subMsg)
 	if err != nil {
-		logger.Error("acct: SubmitObservationToContract failed to send broadcast", zap.Error(err))
-	} else {
-		if txResp.TxResponse == nil {
-			return txResp, fmt.Errorf("txResp.TxResponse is nil")
-		}
-		if strings.Contains(txResp.TxResponse.RawLog, "out of gas") {
-			return txResp, fmt.Errorf("out of gas: %s", txResp.TxResponse.RawLog)
-		}
-
-		out, err := wormchainConn.BroadcastTxResponseToString(txResp)
-		if err != nil {
-			logger.Error("acct: SubmitObservationToContract failed to parse broadcast response", zap.Error(err))
-		} else {
-			logger.Debug("acct: in SubmitObservationToContract, done sending broadcast", zap.String("resp", out))
-		}
+		return txResp, fmt.Errorf("failed to send broadcast: %w", err)
 	}
-	return txResp, err
-}
 
-// CheckSubmitObservationResult() is a free function that returns true if the observation has already been committed
-// or false if we need to wait for the commit. An error is returned when appropriate.
-func CheckSubmitObservationResult(txResp *sdktx.BroadcastTxResponse) (bool, error) {
 	if txResp == nil {
-		return false, fmt.Errorf("txResp is nil")
+		return txResp, fmt.Errorf("sent broadcast but returned txResp is nil")
 	}
+
 	if txResp.TxResponse == nil {
-		return false, fmt.Errorf("txResp does not contain a TxResponse")
+		return txResp, fmt.Errorf("sent broadcast but returned txResp.TxResponse is nil")
 	}
+
 	if txResp.TxResponse.RawLog == "" {
-		return false, fmt.Errorf("RawLog is not set")
+		return txResp, fmt.Errorf("sent broadcast but raw_log is not set, unable to analyze the result")
 	}
-	if strings.Contains(txResp.TxResponse.RawLog, "execute wasm contract failed") {
-		if strings.Contains(txResp.TxResponse.RawLog, "already committed") {
-			return true, nil
 
-		}
+	if strings.Contains(txResp.TxResponse.RawLog, "out of gas") {
+		return txResp, fmt.Errorf("out of gas: %s", txResp.TxResponse.RawLog)
+	}
 
-		// TODO Need to test this, requires multiple guardians.
-		if strings.Contains(txResp.TxResponse.RawLog, "duplicate signature") {
-			return false, nil
-		}
+	if strings.Contains(txResp.TxResponse.RawLog, "failed to execute message") {
+		return txResp, fmt.Errorf("failed to submit observations: %s", txResp.TxResponse.RawLog)
+	}
+
+	logger.Debug("acct: in SubmitObservationsToContract, done sending broadcast", zap.String("resp", wormchainConn.BroadcastTxResponseToString(txResp)))
+	return txResp, nil
+}
+
+// TODO: Once the issue with the junk characters is sorted out, drop the logger parameter.
 
-		return false, fmt.Errorf(txResp.TxResponse.RawLog)
+// GetObservationResponses is a free function that extracts the observation responses from a transaction response.
+// It assumes the transaction response is valid (SubmitObservationsToContract() did not return an error).
+func GetObservationResponses(logger *zap.Logger, txResp *sdktx.BroadcastTxResponse, numExpected int) (ObservationResponses, error) {
```

Comment:
I think the function would be more flexible if numExpected was handled by the caller

Reply:
Done.